pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -708,3 +715,93 @@ uint8_t semtech_loramac_recv(semtech_loramac_t *mac)
 
     return ret;
 }
+
+#ifdef MODULE_PERIPH_EEPROM
+void semtech_loramac_save_config(semtech_loramac_t *mac)
+{
+    DEBUG("[semtech-loramac] saving configuration on EEPROM\n");
+
+    uint8_t magic[] = SEMTECH_LORAMAC_EEPROM_MAGIC;
+    size_t pos = SEMTECH_LORAMAC_EEPROM_START;
+    pos += eeprom_write(pos, magic, SEMTECH_LORAMAC_EEPROM_MAGIC_LEN);
+
+    /* Store EUIs, Keys and device address */
+    pos += eeprom_write(pos, mac->deveui, LORAMAC_DEVEUI_LEN);
+    pos += eeprom_write(pos, mac->appeui, LORAMAC_APPEUI_LEN);
+    pos += eeprom_write(pos, mac->appkey, LORAMAC_APPKEY_LEN);
+    pos += eeprom_write(pos, mac->appskey, LORAMAC_APPSKEY_LEN);
+    pos += eeprom_write(pos, mac->nwkskey, LORAMAC_NWKSKEY_LEN);
+    pos += eeprom_write(pos, mac->devaddr, LORAMAC_DEVADDR_LEN);
+
+    /* uplink counter, mainly used for ABP activation */
+    uint8_t counter[4];
+    mutex_lock(&mac->lock);
+    MibRequestConfirm_t mibReq;
+    mibReq.Type = MIB_UPLINK_COUNTER;
+    LoRaMacMibGetRequestConfirm(&mibReq);
+    *(uint32_t *)counter = mibReq.Param.UpLinkCounter;
+    mutex_unlock(&mac->lock);
+    pos += eeprom_write(pos, counter, 4);
+}
+
+void _set_uplink_counter(semtech_loramac_t *mac, uint8_t *counter)
+{
+    DEBUG("[semtech-loramac] uplink counter: %lu\n", *(uint32_t *)counter);
+
+    mutex_lock(&mac->lock);
+    MibRequestConfirm_t mibReq;
+    mibReq.Type = MIB_UPLINK_COUNTER;
+    mibReq.Param.UpLinkCounter = *(uint32_t *)counter;
```

Comment:
this breaks the strict aliasing rule (as seen in Murdock output). Also the way how this variable is read depends on the endianess of the CPU.